components/src/dynamo/sglang/llm_engine.py

```
@@ -255,7 +259,44 @@ async def abort(self, context: Context) -> None:
+                try:
+                    scheduler = getattr(
+                        self.engine.tokenizer_manager, "scheduler", None
+                    )
+                    if scheduler is not None and scheduler.is_idle():
+                        logger.info("All in-flight SGLang requests drained")
+                        break
+                    total = getattr(scheduler, "get_total_usage", lambda: None)() or 0
+                    logger.debug(
+                        "Waiting for SGLang scheduler to drain (usage=%d)", total
+                    )
+                except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)